java/server/src/org/openqa/selenium/grid/distributor/local/LocalDistributor.java

```diff
@@ -150,6 +159,100 @@ public CreateSessionResponse newSession(HttpRequest request)
     }
   }
 
+  /**
+   * Takes a Stream of Hosts, along with the Capabilities of the current request, and prioritizes the
+   * request by removing Hosts that offer Capabilities that are more rare. e.g. if there are only a
```

Comment:
I suspect that this will ultimately causes problems, since it implies that if a host offers both chrome and edge, it will _never_ run an edge test, no matter what. It would seem like a prioritised list is what we really want here. We can iterate later.

Reply:
Agreed. I'll think about it some more. The next iteration might be as simple as sorting those to the bottom of the list, rather than removing them, but there might be better methods still.

Thanks for landing my first code commit!